### ISSUE TITLE
[Feature Request] memory issue / make Run more efficient

### ISSUE THREAD

**Issue by PaulLerner** [AmenRa/ranx#48]
Hi Elias,

**Is your feature request related to a problem? Please describe.**
I've noticed that `Run` (and I guess also `Qrels`) consume a lot of memory (RAM) compared to standard python `dict`, e.g. a few GB instead of a few 100s of MB. This gets problematic for somewhat large datasets (e.g. 1M queries)

**Describe the solution you'd like**
I guess it's related to Numba representation? I've no clue on how to make it more efficient, sorry :)

**Reproduce**
Just open your system monitor and see how the memory grows.

```py
In [1]: import ranx
# this weighs only a few 100s of MB
In [2]: run_d = {str(i): {str(j): 0.0 for j in range(100)} for i in range(100000)}
# this grows to a few GB
In [3]: run_r = ranx.Run(run_d)
```

Best,

Paul

**Comment by AmenRa** [comment]
Fixed in `v0.3.15`.

**Comment by AmenRa** [comment]
Fixed in `v0.3.15`.
